Mathlib/Algebra/MvPolynomial/SchwartzZippel.lean

```diff
@@ -0,0 +1,144 @@
+            _ ≤ #{xₜ ∈ tail S ^^ n | eval xₜ pₖ = 0} / ∏ i, (#S (.succ i) : ℚ≥0) :=
+              mul_div_mul_left_le (by positivity) (by positivity)
+            _ ≤ ∑ i, (pₖ.degreeOf i / #S (.succ i) : ℚ≥0) := schwartz_zippel hpₖ₀ _
+            _ ≤ ∑ i, (p.degreeOf (.succ i) / #S (.succ i) : ℚ≥0) := by
+              gcongr; exact degreeOf_coeff_finSuccEquiv ..
+        · -- We bound the second set by noting that if `x` is in it, then `x₀` is the root of
+          -- the univariate polynomial`pₓ` obtained by evaluating each (multivariate polynomial)
+          -- coefficient at `xₜ`. Since `pₓ` has degree `k`, there are at most `k` such `x₀` for
+          -- each `xₜ`, which gives the result.
+          calc
+            #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ ≠ 0} / ∏ i, (#S i : ℚ≥0)
+              ≤ ↑(p.degreeOf 0 * ∏ i, #S (.succ i)) / ∏ i, (#S i : ℚ≥0) := ?_
+            _ = p.degreeOf 0 * (∏ i, #S (.succ i)) / (#S 0 * ∏ i, #S (.succ i)) := by
+              norm_cast; rw [prod_univ_succ]
+            _ ≤ (p.degreeOf 0 / #S 0 : ℚ≥0) := mul_div_mul_right_le (by positivity) (by positivity)
+          gcongr
+          calc
+            #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ ≠ 0}
+              = #{x ∈ S ^^ (n + 1) | eval (tail x) pₖ ≠ 0 ∧ eval x p = 0} := by simp_rw [and_comm]
+            _ = #({xₜ ∈ tail S ^^ n | eval xₜ pₖ ≠ 0}.biUnion fun xₜ ↦ image (fun x₀ ↦ (x₀, xₜ))
+                  {x₀ ∈ S 0 | eval (cons x₀ xₜ) p = 0}) := by
+              rw [← filter_filter, filter_piFinset_eq_map_consEquiv S (fun r ↦ eval r pₖ ≠ 0),
+                filter_map, card_map, product_eq_biUnion_right, filter_biUnion]
+              simp [Function.comp_def, Finset.filter_image, filter_filter]
+              rfl
+            _ ≤ ∑ xₜ ∈ tail S ^^ n with eval xₜ pₖ ≠ 0,
+                  #image (fun x₀ ↦ (x₀, xₜ)) {x₀ ∈ S 0 | eval (cons x₀ xₜ) p = 0} := card_biUnion_le
+            _ ≤ ∑ xₜ ∈ tail S ^^ n with eval xₜ pₖ ≠ 0, #{x₀ ∈ S 0 | eval (cons x₀ xₜ) p = 0} := by
+              gcongr; exact card_image_le
+            _ ≤ ∑ xₜ ∈ tail S ^^ n with eval xₜ pₖ ≠ 0, p.degreeOf 0 := ?_
+            _ ≤ ∑ _xₜ ∈ tail S ^^ n, p.degreeOf 0 := by gcongr; exact filter_subset ..
+            _ = p.degreeOf 0 * ∏ i, #S (.succ i) := by simp [mul_comm, tail]
+          gcongr with xₜ hxₜ
+          simp at hxₜ
+          set pₓ := p'.map (eval xₜ) with hpₓ
+          have hpₓdeg : pₓ.natDegree = k := by
+            rw [hpₓ, hk, Polynomial.natDegree_map_of_leadingCoeff_ne_zero (eval xₜ) hxₜ.2]
+          have hpₓ₀ : pₓ ≠ 0 := fun h ↦ hxₜ.2 <| by
+            rw [hpₖ, Polynomial.leadingCoeff, ← hk, ← hpₓdeg, h, Polynomial.natDegree_zero,
+              ← Polynomial.coeff_map, ← hpₓ, h, Polynomial.coeff_zero]
+          calc
+            #{x₀ ∈ S 0 | eval (cons x₀ xₜ) p = 0} ≤ #pₓ.roots.toFinset := by
+              gcongr
+              simp (config := { contextual := true}) [subset_iff, eval_eq_eval_mv_eval', pₓ, hpₓ₀]
```

Comment:
```suggestion
              simp (config := {contextual := true}) [subset_iff, eval_eq_eval_mv_eval', pₓ, hpₓ₀]
```
I don't really mind where the spaces are, but let's be consistent on both sides!

Reply:
I knew something was itching me with this simp call but I couldn't see what!